pkg/apis/core/validation/shoot.go

```diff
@@ -761,11 +857,35 @@ func fieldNilOrEmptyString(field *string) bool {
 	return field == nil || len(*field) == 0
 }
 
-func validateNetworking(networking core.Networking, fldPath *field.Path) field.ErrorList {
+func validateNetworking(networking *core.Networking, workerlessShoot bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if len(networking.Type) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("type"), "networking type must be provided"))
+	if workerlessShoot {
```

Comment:
Can you extract this to another function `validateForbiddenNetworkSettingsForWorkerless`?
In that function you can first check if `networking` passed to it is nil and exit early rather than in the `else` clause.

So in the end you have 3 sections:
- explicit checks for subset of fields which are forbidden for workerless shoots
- explicit checks for settings that are forbidden for non-workerless shoots (networking to not be nil, networing.type to not be nil)
- common set of validations that needs to be applied to all shoots.

Other questions:
1. In the markdown you marked networking.Nodes, networking.Pods as forbidden for workerless but i do not see that check being done.

Reply:
The validations for workerless and shoot with workers are not mutually exclusive, so there will be a lot of duplication. I suggest to not do that.